tasktiger/worker.py

```diff
@@ -214,7 +215,33 @@ def _poll_for_queues(self) -> None:
         """
         if not self._did_work:
             time.sleep(self.config["POLL_TASK_QUEUES_INTERVAL"])
-        self._refresh_queue_set()
+
+        if self._should_poll_for_queues():
+            self._refresh_queue_set()
+            self._next_forced_queue_poll = time.monotonic() + 3600
+
+    def _should_poll_for_queues(self) -> bool:
+        if not self.only_queues:
+            return True
+
+        queue_set_token = self._get_queue_set_token()
+
+        if queue_set_token != self._queue_set_token:
+            self._queue_set_token = queue_set_token
+            return True
+
+        if time.monotonic() > self._next_forced_queue_poll:
+            self.log.info("Forcing a queue poll due to inactivity.")
+            return True
+
+        return False
+
+    def _get_queue_set_token(self) -> str:
+        keys = sorted(
+            self._key("queue_token", queue) for queue in self.only_queues
```

Comment:
If we don't use `only_queues`, we'd poll the same way we do now (every `POLL_TASK_QUEUES_INTERVAL`)

Reply:
> What if we don't use only_queues? Would we then only poll every hour?
The way it is currently implemented, this wouldn't work if we process a sub-queue (e.g. "foo.bar"), right?

Correct, but all we need to do is take a root of each queue from `self.only_queues` to get it work.

Reply:
> If we don't use only_queues, we'd poll the same way we do now (every POLL_TASK_QUEUES_INTERVAL)

Ah, because `_should_poll_for_queues` returns `True` if `not self.only_queues`.